src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -61,13 +61,27 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
                 return issuerValidationResult.UnwrapError().AddStackFrame(StackFrames.IssuerValidationFailed);
             }
 
-            var signatureValidationResult = ValidateSignature(samlToken, validationParameters, callContext);
+            ValidationResult<SecurityKey> signatureValidationResult = ValidateSignature(samlToken, validationParameters, callContext);
+
             if (!signatureValidationResult.IsValid)
             {
                 StackFrames.SignatureValidationFailed ??= new StackFrame(true);
                 return signatureValidationResult.UnwrapError().AddStackFrame(StackFrames.SignatureValidationFailed);
             }
 
+            ValidationResult<ValidatedSigningKeyLifetime> issuerSigningKeyValidationResult = validationParameters.IssuerSigningKeyValidator(
+                samlToken.SigningKey,
+                samlToken,
+                validationParameters,
+                null,
+                callContext);
+
+            if (!issuerSigningKeyValidationResult.IsValid)
+            {
+                StackFrames.IssuerSigningKeyValidationFailed ??= new StackFrame(true);
```

Comment:
Is it possible for this to be set already?

Reply:
After the first time it would. However, we are moving to use the simplified format when caching frames.
I will be raising a PR soon replacing all of these `new StackFrame(true)` instances.